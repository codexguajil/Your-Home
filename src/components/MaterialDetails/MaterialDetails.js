import React, { Component } from 'react';
import { Modal, Button, Tabs } from 'antd';

class MaterialDetails extends Component {
  state = { visible: false };

  showModal = () => {
    this.setState({ visible: true });
  }

  handleOk = () => {
    this.setState({ visible: false });
  }

  handleCancel = () => {
    this.setState({ visible: false });
  }

  renderDetails = () => {
    const TabPane = Tabs.TabPane;
    const { brand, manual_url, model_number, name, notes, quantity, unit_price, vendor } = this.props;
    return(
      <Tabs defaultActiveKey="1">
        <TabPane tab="General" key="1">
          <p>Name: {name}</p>
          <p>Brand: {brand}</p>
          <p>Quantity: {quantity}</p>
          {manual_url && 
            <p>Link to manual: {manual_url}</p>
          }
        </TabPane>
        <TabPane tab="Product Details" key="2">
          <p>Model Number: {model_number}</p>
          <p>Vendor: {vendor}</p>
          <p>Price: {unit_price}</p>
        </TabPane>
        <TabPane tab="Notes" key="3">
          <p>{notes}</p>
        </TabPane>
      </Tabs>
    );
  }

  render() {
    return (
      <div>
        <Button type="link" onClick={this.showModal}>
          View Details
        </Button>
        <Modal
          title={this.props.name}
          visible={this.state.visible}
          onOk={this.handleOk}
          onCancel={this.handleCancel}
        >
          <div>
            {this.renderDetails()}
          </div>
        </Modal>
      </div>
    );
  }
}

export default MaterialDetails;