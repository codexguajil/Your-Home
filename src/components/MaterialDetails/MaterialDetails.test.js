import React from 'react';
import { shallow } from 'enzyme';
import MaterialDetails from './MaterialDetails';

describe('MaterialDetails', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MaterialDetails/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state visible when showModal is called', () => {
    wrapper.setState({ visible: false })
    wrapper.instance().showModal();
    expect(wrapper.state('visible')).toBe(true);
  });

  it('should update state visible when handleOK is called', () => {
    wrapper.setState({ visible: true })
    wrapper.instance().handleOk();
    expect(wrapper.state('visible')).toBe(false);
  });
  
  it('should update state visible when handleCancel is called', () => {
    wrapper.setState({ visible: true })
    wrapper.instance().handleCancel();
    expect(wrapper.state('visible')).toBe(false);
  });

});