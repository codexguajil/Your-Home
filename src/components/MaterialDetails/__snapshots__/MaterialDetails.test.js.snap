// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialDetails should match snapshot 1`] = `
<div>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="link"
  >
    View Details
  </Button>
  <Modal
    cancelButtonDisabled={false}
    confirmLoading={false}
    maskTransitionName="fade"
    okButtonDisabled={false}
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <div>
      <Tabs
        defaultActiveKey="1"
        hideAdd={false}
        tabPosition="top"
      >
        <TabPane
          key="1"
          placeholder={null}
          tab="General"
        >
          <p>
            Name: 
          </p>
          <p>
            Brand: 
          </p>
          <p>
            Quantity: 
          </p>
        </TabPane>
        <TabPane
          key="2"
          placeholder={null}
          tab="Product Details"
        >
          <p>
            Model Number: 
          </p>
          <p>
            Vendor: 
          </p>
          <p>
            Price: 
          </p>
        </TabPane>
        <TabPane
          key="3"
          placeholder={null}
          tab="Notes"
        >
          <p>
            Notes: 
          </p>
        </TabPane>
      </Tabs>
    </div>
  </Modal>
</div>
`;
